### PR TITLE
add doctests

### PR DESCRIPTION
**Pull request by zqigolden**

### CODE REVIEW

File: deer/deer_iter.py [hunk 1]
```diff
@@ -68,6 +68,44 @@ def deer_iteration(
...
+    ...     inv_lin_params=inv_lin_params,
+    ...     yinit_guess=yinit_guess,
+    ...     max_iter=100,
+    ...     memory_efficient=False,
+    ...     clip_ytnext=False
+    ... )
+
+    >>> print(result)
+    [[0.4]
+     [0.3]]
```
Comment (mfkasim1) [inline deer/deer_iter.py:108]: This is an internal function, so we don't need an example for this

File: deer/fseq1d.py [hunk 2]
```diff
@@ -43,6 +43,32 @@ def seq1d(func: Callable[[jnp.ndarray, Any, Any], jnp.ndarray],
...
+    >>> from fseq1d import seq1d, Sequential
+    
+    >>> def func(y, x, params):
+    ...     return y ** 2 + x * params[0]
+    
+    >>> y0 = jnp.array([0.0])
+    >>> xinp = jnp.linspace(0, 1, 10).reshape(-1, 1)
+    >>> params = jnp.array([0.5])
+    
+    >>> y = seq1d(func, y0, xinp, params, method=Sequential())
```
Comment (mfkasim1) [inline deer/fseq1d.py:60]: Could you use `seq1d.Sequential` here?

File: deer/maths.py [hunk 3]
```diff
@@ -33,6 +33,26 @@ def matmul_recursive(mats: jnp.ndarray, vecs: jnp.ndarray, y0: jnp.ndarray) -> j
...
+    >>> vecs = jnp.array([
+    ...     [1.0, 1.0],
+    ...     [0.0, 0.0]
+    ... ])
+    >>> y0 = jnp.array([1.0, 2.0])
+    >>> matmul_recursive(mats, vecs, y0)
+    Array([[1. , 2. ],
+           [2. , 3. ],
+           [2.5, 2.5]], dtype=float64)
+
```
Comment (mfkasim1) [inline deer/maths.py:55]: This is an internal function, so we don't need an example here

**Review by mfkasim1 — CHANGES_REQUESTED** [review]

**Comment by mfkasim1** [comment]
Also, can you add doctest in CI with this PR?

File: deer/maths.py (hunk 3 above)
Comment (zqigolden) [reply deer/maths.py]: It has been removed

File: deer/fseq1d.py (hunk 2 above)
Comment (zqigolden) [reply deer/fseq1d.py]: Sure, changed

**Comment by zqigolden** [comment]
https://github.com/machine-discovery/deer/pull/23#issuecomment-2158970535

Yes, it has added to ci now

**Review by mfkasim1 — APPROVED** [review]

**Pull request merged**